tools/schemapi/schemapi.py

```diff
@@ -346,7 +362,7 @@ def __init__(self, obj: "SchemaBase", err: jsonschema.ValidationError) -> None:
         super().__init__(**err._contents())
         self.obj = obj
         self._errors: GroupedValidationErrors = getattr(
-            err, "_all_errors", {err.json_path: [err]}
+            err, "_all_errors", {_json_path(err): [err]}
```

Comment:
```suggestion
            err, "_all_errors", {getattr(err, "json_path", _json_path(err)): [err]}
```
I'd suggest to consistently use the `.json_path` attribute if it's available and only fall back for older versions.